DO NOT MERGE

jchunkins opened the pull request:
DO NOT MERGE

jchunkins commented on stable/logdna-agent/README.md:
```diff
@@ -19,7 +19,111 @@ This chart deploys LogDNA collector agents to all nodes in your cluster. Logs wi
 ### Pod Security Policies
 
-The namespace you deploy into must allow the container to run as root, and must be able to access hostPath volumes. 
+LogDNA-agent containers needs to mount host volumes to read the logs for ingestion. Therefore, this chart requires a PodSecurityPolicy that permits host volume access to be bound to the target namespace prior to installation. If the default PodSecurityPolicy on your cluster is not restrictive then this step is not needed.
```
```suggestion
LogDNA-agent containers needs to mount host volumes to read the logs for ingestion, and needs to run as root. This chart requires a PodSecurityPolicy (with host volume and root access) to be bound to the target namespace prior to installation. If the `default` PodSecurityPolicy on your namespace is not restrictive then this step is not needed.
```

jeyaramashok commented:
I used these two charts as references to come up with section on PSP

- https://github.com/IBM/charts/blob/master/stable/ibm-mobilefoundation-dev/README.md#podsecuritypolicy-requirements
- https://github.com/IBM/charts/tree/master/community/openwhisk#podsecuritypolicy-requirements

jchunkins commented on stable/logdna-agent/README.md:
```diff
@@ -19,7 +19,111 @@ This chart deploys LogDNA collector agents to all nodes in your cluster. Logs wi
+
+If the default is restrictive, In ICP you can create a new namespace with either a predefined PodSecurityPolicy.
+
+- Predefined PodSecurityPolicy name: [ibm-anyuid-hostpath-psp](https://github.com/IBM/cloud-pak/blob/master/spec/security/psp/README.md)
```
```suggestion
```

jchunkins commented on stable/logdna-agent/README.md:
````diff
@@ -19,7 +19,111 @@ This chart deploys LogDNA collector agents to all nodes in your cluster. Logs wi
+- '*'
+```
+
+- Create a [ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings)
````
```suggestion
- Create a [ClusterRole](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#default-roles-and-role-bindings). Save the following contents into a file (e.g. psp.yaml)
```

jchunkins commented on stable/logdna-agent/README.md:
````diff
@@ -19,7 +19,111 @@ This chart deploys LogDNA collector agents to all nodes in your cluster. Logs wi
+    - create
+    - delete
+```
+
````
```suggestion
kubectl command:
```
Jey... put this command here in markdown triple tilde quotes:

kubectl create -f psp.yaml

jchunkins commented on stable/logdna-agent/README.md:
```diff
@@ -19,7 +19,111 @@ This chart deploys LogDNA collector agents to all nodes in your cluster. Logs wi
+
+Create a ClusterRoleBinding which binds ClusterRole created in previous step to default service account of the namespace.
+
+kubectl command:
```
```suggestion
kubectl command (replace `<namespace>` with your namespace):
```

jchunkins commented on stable/logdna-agent/README.md:
````diff
@@ -19,7 +19,111 @@ This chart deploys LogDNA collector agents to all nodes in your cluster. Logs wi
+    namespace: <namespace>
+```
+
+Alternatively, a cluster admin can create a [service account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) for LogDNA itself and bind the ClusterRole to it.
````
```suggestion
Alternatively, a cluster admin can create a [service account](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) for LogDNA itself and bind the ClusterRole to it. Save the following contents into a file (e.g. sa.yaml):
```

jchunkins commented on stable/logdna-agent/README.md:
````diff
@@ -19,7 +19,111 @@ This chart deploys LogDNA collector agents to all nodes in your cluster. Logs wi
+    name: logdna-serviceaccount
+    namespace: <namespace>
+```
 
````
```suggestion
kubectl command:
```

Jey... put this command here in markdown triple tilde quotes:

kubectl create -f sa.yaml

jeyaramashok commented:
@jchunkins I have fixed the code review suggestions, pls take a look when you get a chance.

jchunkins committed:
First pass at applying ibm suggested changes

jchunkins committed:
various changes

- update icon url
- use correct license
- update default "empty" values
- add metadata file
- update notes.txt

jchunkins committed:
add support link
